src/python/bot/fuzzers/blackbox/engine.py

```diff
@@ -0,0 +1,171 @@
+    --testcase_path: If specified, a path to a test case to reproduce. Output
+                     should be written directly to stdout or stderr.
+  """
+
+  @property
+  def name(self):
+    return 'blackbox'
+
+  def prepare(self, corpus_dir, target_path, build_dir):  # pylint: disable=unused-argument
+    """Prepare for a fuzzing session by generating options.
+
+    Though blackbox fuzzers follow the engine interface, they must be launched
+    in a different manner from most other engine fuzzers. Instead of running a
+    target directly, these fuzzers tend to be wrapper scripts which generate
+    test cases and pass them to a binary that is managed by the infrastructure.
+
+    Args:
+      corpus_dir: The main corpus directory.
+      target_path: Path to the target.
+      build_dir: Path to the build directory.
+    Returns:
+      A FuzzOptions object.
+    """
+    app_path = environment.get_value('APP_PATH')
+    app_args = testcase_manager.get_command_line_for_application(
+        get_arguments_only=True).strip()
+    arguments = [
+        f'--app_path={app_path}',
+        f'--app_args={app_args}',
+        f'--input_dir={corpus_dir}',
+    ]
+    return engine.FuzzOptions(corpus_dir, arguments, {})
+
+  def fuzz(self, target_path, options, reproducers_dir, max_time):
+    """Run a fuzzing session.
+    Args:
+      target_path: Path to the target.
+      options: The FuzzOptions object returned by prepare().
+      reproducers_dir: The directory to put reproducers in when crashes
+          are found.
+      max_time: Maximum allowed time for the fuzzing to run.
+   Returns:
+      A FuzzResult object.
+    """
+    os.chmod(target_path, 0o775)
+
+    args = options.arguments
+    args.append(f'--output_dir={reproducers_dir}')
+
+    runner = new_process.UnicodeProcessRunner(target_path)
+    result = runner.run_and_wait(timeout=max_time, additional_args=args)
+
+    crashes = []
+    for testcase_path in os.listdir(reproducers_dir):
+      if not testcase_path.startswith(TESTCASE_PREFIX):
+        continue
+
+      output_path = OUTPUT_PREFIX + testcase_path[len(TESTCASE_PREFIX):]
+      absolute_output_path = os.path.join(reproducers_dir, output_path)
+
+      # If no output was written for a test case, skip it.
+      if not os.path.exists(absolute_output_path):
+        continue
+
+      with open(absolute_output_path, 'r', errors='replace') as handle:
+        output = handle.read()
+
+      # Filter obviously non-crashing test cases. Crashes still follow the
+      # normal flow in fuzz task to ensure that the state should not be ignored
+      # for other reasons, but we don't want to log every test case for the
+      # fuzzers that don't do their own crash processing.
+      state = stack_analyzer.get_crash_data(output)
+      if not state.crash_type:
+        continue
+
+      crash = engine.Crash(
+          os.path.join(reproducers_dir, testcase_path), output, [],
```

Comment:
Why [] and not args ?

move os.path.join(reproducers_dir, testcase_path) into a local

Reply:
They're currently unused in reproduce, but they could be useful in the future. I'll make the change here. Also moved that expression to a local.